### PR TITLE
[MRG] Update legacy json and param conversion function for new json format

### PR DESCRIPTION
**Pull request by gtdang**
This changes the convert_to_hdf5 function to convert_to_json. 
* The function converts legacy flat json and param files to the new hierarchical json format. 
* It also adds network connectivity. By default it uses the Jones 2009 model for the connectivity but supports the other pre-defined network models. No network connectivity can be added by supplying None.

Question:
* Should the default be None instead of the Jones 2009 for adding network connectivity?

### CODE REVIEW

File: doc/whats_new.rst [hunk 1]
```diff
@@ -58,6 +55,9 @@ Changelog
 - Added feature to read/write :class:`~hnn_core.Network` configurations to
   json, by `George Dang`_ and `Rajat Partani`_ in :gh:`757`
 
+- Add feature to convert legacy param and json files to new json format,
```
Comment (jasmainak) [inline doc/whats_new.rst:58]: you should give link to function etc.

File: hnn_core/params.py [hunk 2]
```diff
@@ -657,27 +657,50 @@ def compare_dictionaries(d1, d2):
...
 
-def convert_to_hdf5(params_fname, out_fname, include_drives=True,
-                    overwrite=True, write_output=False):
-    """Converts json or param format to hdf5
+def convert_to_json(params_fname,
+                    out_fname,
+                    network_connectivity='jones_2009_model',
+                    include_drives=True,
+                    overwrite=True):
+    """Converts legacy json or param format to hierarchical json format
```
Comment (jasmainak) [inline hnn_core/params.py:665]: sorry I missed the discussion. Do you guys have a specification of the hierarchical json format? how will you store connectivity etc? Or is it an intermediate format before going to hdf5?

Comment (gtdang) [reply hnn_core/params.py:665]: The format is Network object converted to a dictionary format. So the connectivities are just the same hierarchical dictionary format as the Network Object. Pretty much everything is preserved except numpy arrays need to be converted to lists when written to json and converted back upon read-in. 

This is not an intermediate format before going to hdf5. All the network connections, drives, cell parameters can be saved and loaded with this format. 

What is not included with this format is the ability to save outputs (the optional recordings during simulation saved to the Network). [Is there a better term to use than "outputs"?] We plan to use the hdf5 format for the saving of network outputs. For that hdf5 file, a version of this json will be saved as a single dataset in the hdf5 (A string json binarized), that way the network configurations are paired with the outputs.

**Comment by jasmainak** [comment]
Do these new json format files work with the GUI?

**Comment by gtdang** [comment]
> Do these new json format files work with the GUI?

No, not yet.

File: hnn_core/params.py [hunk 3]
```diff
@@ -662,27 +662,50 @@ def compare_dictionaries(d1, d2):
...
 
     Parameters
     ----------
     params_fname : str or Path
         Path to file
     out_fname: str
         Path to output
+    network_connectivity: str or None, default:' jones_2009_model'
```
Comment (ntolley) [inline hnn_core/params.py:678]:
```suggestion
    model_template: str or None, default:' jones_2009_model'
```
?

File: hnn_core/tests/test_params.py [hunk 4]
```diff
@@ -74,29 +79,140 @@ def test_base_params():
...
+        # Write json and check if constructed network is equal
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath,
+                        network_connectivity='calcium_model')
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
+
+    def test_no_network_connectivity(self, tmp_path):
+        """Tests conversion with no network connectivity model"""
```
Comment (ntolley) [inline hnn_core/tests/test_params.py:149]: good edge case to check!

File: hnn_core/tests/test_params.py [hunk 5]
```diff
@@ -74,29 +79,140 @@ def test_base_params():
...
+                             add_drives_from_params=True,
+                             )
+
+        # Write json and check if constructed network is equal
+        outpath = Path(tmp_path, 'default.json')
+        convert_to_json(self.path_default,
+                        outpath,
+                        network_connectivity=None)
+        net_json = read_network_configuration(outpath)
+        assert net_json == net_params
```
Comment (ntolley) [inline hnn_core/tests/test_params.py:161]:
```suggestion
        assert net_json == net_params
        assert len(net_json.connectivity) == len(net_param.connectivity) == 0
```

File: hnn_core/params.py (hunk 3 above)
Comment (gtdang) [reply hnn_core/params.py]: Made this argument name change. Thanks!

File: hnn_core/tests/test_params.py (hunk 5 above)
Comment (gtdang) [reply hnn_core/tests/test_params.py:161]: I added this check but instead of 0, I'm using 22. This is because there are still 22 connections due to external drives.

**Review by ntolley — APPROVED** [review]
All good on my end, thanks @gtdang !!
